diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -505,34 +507,41 @@ int DifferentialFunctionComparator::cmpAllocs(const CallInst *CL,
 
 /// Check if the given operation can be ignored (it does not affect semantics)
 /// for control flow only diffs.
-bool DifferentialFunctionComparator::mayIgnore(const User *Inst) const {
-    if (config.ControlFlowOnly)
-        return isa<AllocaInst>(Inst) || isCast(Inst);
-    else {
-        if (isa<AllocaInst>(Inst))
-            return true;
+bool DifferentialFunctionComparator::mayIgnore(const User *Inst,
+                                               bool addToIgnored) const {
```

Comment:
Refactor this to `bool maySkipInstruction(const Instruction *Inst)`. The function should always insert into the `IgnoredInstructions` map if it returns true.